class org.junit.experimental.results.FailureList extends java.lang.Object
{
    private final java.util.List failures;

    public void <init>(java.util.List)
    {
        org.junit.experimental.results.FailureList r0;
        java.util.List r1;

        r0 := @this: org.junit.experimental.results.FailureList;

        r1 := @parameter0: java.util.List;

        specialinvoke r0.<java.lang.Object: void <init>()>();

        r0.<org.junit.experimental.results.FailureList: java.util.List failures> = r1;

        return;
    }

    public org.junit.runner.Result result()
    {
        org.junit.experimental.results.FailureList r0;
        org.junit.runner.notification.RunListener r1;
        java.util.Iterator r2;
        org.junit.runner.notification.Failure r3;
        org.junit.runner.Result $r4;
        java.util.List $r5;
        boolean $z0;
        java.lang.Object $r6;
        java.lang.Exception $r7;
        java.lang.RuntimeException $r8;

        r0 := @this: org.junit.experimental.results.FailureList;

        $r4 = new org.junit.runner.Result;

        specialinvoke $r4.<org.junit.runner.Result: void <init>()>();

        r1 = virtualinvoke $r4.<org.junit.runner.Result: org.junit.runner.notification.RunListener createListener()>();

        $r5 = r0.<org.junit.experimental.results.FailureList: java.util.List failures>;

        r2 = interfaceinvoke $r5.<java.util.List: java.util.Iterator iterator()>();

     label1:
        $z0 = interfaceinvoke r2.<java.util.Iterator: boolean hasNext()>();

        if $z0 == 0 goto label5;

        $r6 = interfaceinvoke r2.<java.util.Iterator: java.lang.Object next()>();

        r3 = (org.junit.runner.notification.Failure) $r6;

     label2:
        virtualinvoke r1.<org.junit.runner.notification.RunListener: void testFailure(org.junit.runner.notification.Failure)>(r3);

     label3:
        goto label1;

     label4:
        $r7 := @caughtexception;

        $r8 = new java.lang.RuntimeException;

        specialinvoke $r8.<java.lang.RuntimeException: void <init>(java.lang.String)>("I can\'t believe this happened");

        throw $r8;

     label5:
        return $r4;

        catch java.lang.Exception from label2 to label3 with label4;
    }
}
